modules/markup/external/external.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package external
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/markup"
+	"code.gitea.io/gitea/modules/setting"
+
+	gouuid "github.com/satori/go.uuid"
+)
+
+// RegisterParsers registers all supported third part parsers according settings
+func RegisterParsers() {
+	for _, render := range setting.ExternalMarkupRenders {
+		if render.Enabled && render.Command != "" && len(render.FileExtensions) > 0 {
+			markup.RegisterParser(&Parser{render})
+		}
+	}
+}
+
+// Parser implements markup.Parser for external tools
+type Parser struct {
+	setting.MarkupRender
+}
+
+// Name returns the external tool name
+func (p *Parser) Name() string {
+	return p.MarkupName
+}
+
+// Extensions returns the supported extensions of the tool
+func (p *Parser) Extensions() []string {
+	return p.FileExtensions
+}
+
+// Render renders the data of the document to HTML via the external tool.
+func (p *Parser) Render(rawBytes []byte, urlPrefix string, metas map[string]string, isWiki bool) []byte {
+	var (
+		bs       []byte
+		buf      = bytes.NewBuffer(bs)
+		rd       = bytes.NewReader(rawBytes)
+		commands = strings.Fields(p.Command)
+		args     = commands[1:]
+	)
+
+	if p.IsInputFile {
+		// write to templ file
+		fPath := filepath.Join(os.TempDir(), gouuid.NewV4().String())
```

Comment:
You should check if the file already exists although it is very unlikely to happen. Or as an alternative you could add this check to your `err != nil` check at line 60.